tox.ini

```diff
@@ -31,21 +31,46 @@ deps =
 commands = python -m towncrier.check
 
 [testenv:generatedocs]
-deps = 
+deps =
     mbed-tools-ci-scripts
 commands = generate-docs --output_dir {posargs}
 
 [testenv:generatespdx]
-deps = 
+deps =
     mbed-tools-ci-scripts
 commands = generate-spdx --output-dir {posargs}
 
-[testenv:release]
-passenv =
-    GIT_TOKEN
-    TWINE_USERNAME
-    TWINE_PASSWORD
-    IGNORE_PYPI_TEST_UPLOAD
+[testenv:preprelease]
+usedevelop = True
+allowlist_externals = git
 deps =
     mbed-tools-ci-scripts
-commands = tag-and-release {posargs}
+commands =
+    # Update the target database snapshot with the latest targets
+    python ci_scripts/sync_board_database.py -vvv
+    git add src/mbed_tools/targets/_internal/data/board_database_snapshot.json
+
+    # Generate API docs and add license headers to them
+    generate-docs --output_dir docs/api
+    license-files
+    git add docs/api
+
+    # Generate changelog from news fragments
+    towncrier --yes
+    git add ./CHANGELOG.md
+
+    # Commit and tag new version
```

Comment:
This comment says that there may be some tagging going on, but I don't see `git tag` here. I don't think `bump_version.py` should make the `git tag`, as all other git operations happen here at this abstraction level.